includes/handler.inc

```diff
@@ -24,33 +24,62 @@
  * @see hook_islandora_oai_identify_request_handler()
  */
 function islandora_oai_retrieve_sets($params) {
-  $connection = islandora_get_tuque_connection();
-  $records = array();
-  $filters = array('sameTerm(?model, <info:fedora/islandora:collectionCModel>)');
-  $sparql_query = islandora_oai_construct_sparql_query_for_sets($filters);
-  $total_records = $connection->repository->ri->countQuery($sparql_query, 'sparql');
-  $sparql_query .= <<<EOQ
-  LIMIT {$params['max_records']}
-  OFFSET {$params['token']->deliveredrecords}
+  switch (variable_get('islandora_oai_query_backend', 'sparql')) {
+    case 'sparql':
+      $connection = islandora_get_tuque_connection();
+      $records = array();
+      $filters = array('sameTerm(?model, <info:fedora/islandora:collectionCModel>)');
+      $sparql_query = islandora_oai_construct_sparql_query_for_sets($filters);
+      $total_records = $connection->repository->ri->countQuery($sparql_query, 'sparql');
+      $sparql_query .= <<<EOQ
+
+LIMIT {$params['max_records']}
+OFFSET {$params['token']->deliveredrecords}
 EOQ;
-  $query_results = $connection->repository->ri->sparqlQuery($sparql_query);
-  module_load_include('inc', 'islandora', 'includes/dublin_core');
-  foreach ($query_results as $result) {
-    $object = islandora_object_load($result['object']['value']);
-    $description = FALSE;
-    if (isset($object['DC'])) {
-      $dom = new DOMDocument();
-      $dom->loadXML($object['DC']->content);
-      $description = $dom->saveXML($dom->documentElement);
-    }
-    $pid = $result['object']['value'];
-    $records[$pid] = array(
-      'pid' => $pid,
-      'label' => $result['title']['value'],
-    );
-    if ($description) {
-      $records[$pid]['description'] = $description;
-    }
+      $query_results = $connection->repository->ri->sparqlQuery($sparql_query);
+      foreach ($query_results as $result) {
+        $object = islandora_object_load($result['object']['value']);
+        $description = FALSE;
+        if (isset($object['DC'])) {
+          $description = islandora_oai_get_set_dc_portion($object['DC']);
+        }
+        $pid = $result['object']['value'];
+        $records[$pid] = array(
+          'pid' => $pid,
+          'label' => $result['title']['value'],
+        );
+        if ($description) {
+          $records[$pid]['description'] = $description;
+        }
+      }
+      break;
+
+    case 'solr':
+      $qp = new IslandoraSolrQueryProcessor();
+      $label_field = variable_get('islandora_solr_object_label_field', 'fgs_label_s');
+      $solr_params = array(
+        'limit' => $params['max_records'],
+        'sort' => $label_field,
+      );
+      $qp->buildQuery(format_string('!field:"info:fedora/islandora:collectionCModel"', array(
+        '!field' => variable_get('islandora_oai_content_model_field', 'RELS_EXT_hasModel_uri_ms'),
```

Comment:
Unsure where to go here; it's part of the [OAI handler](https://github.com/Islandora/islandora_oai/blob/634c86e1a78cd7167e705d07bb1ec770fd12d5b1/includes/handler.admin.inc#L42-L48), but technically that handler may not be the one in use. And on top of that, that field does seem pretty superfluous given the Solr variable.

Actually, imma just replace it with the Solr variable since that would make the most sense.